x-pack/plugin/security/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -45,3 +49,15 @@ grant codeBase "${codebase.netty-transport}" {
    // the bug says it only happened rarely, and that its fixed, but apparently it still happens rarely!
    permission java.util.PropertyPermission "sun.nio.ch.bugLevel", "write";
 };
+
+grant codeBase "${codebase.oauth2-oidc-sdk}" {
+    // for JSON serialization based on a shaded GSON dependency
+    permission java.lang.RuntimePermission "accessDeclaredMembers";
+    permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
+};
+
+grant codeBase "${codebase.nimbus-jose-jwt}" {
+    // for JSON serialization based on a shaded GSON dependency
+    permission java.lang.RuntimePermission "accessDeclaredMembers";
+    permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
+};
```

Comment:
Probably due to my lack of understanding: Why do we need these specific grants when we already have the broader grants of the same permission above? Also have we defined these codebases anywhere?

Reply:
> Why do we need these specific grants when we already have the broader grants of the same permission above?

This was a suggestion from @ChrisHegarty -- it should _narrow_ the permission scope for these dependencies, not broaden it.